src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs

```diff
@@ -0,0 +1,541 @@
+		/// </summary>
+		/// <remarks>This will be 0 if the entire window is 'safe' for content.</remarks>
+		public static Thickness WindowInsets => GetWindowInsets();
+
+		private static void OnInsetsChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+		{
+#if WINUI
+			// SafeArea is disabled for WinUI 3 and up as there's no available API for bounds.
+#else
+			if (dependencyObject is FrameworkElement fe)
+			{
+				SafeAreaDetails.GetInstance(fe).OnInsetsChanged((InsetMask)args.OldValue, (InsetMask)args.NewValue);
+			}
+			else
+			{
+#if HAS_UNO // Is building using Uno.UI
+				if (dependencyObject.Log().IsEnabled(LogLevel.Debug))
+				{
+					dependencyObject.Log().LogDebug($"Insets is only supported on FrameworkElement (Found {dependencyObject?.GetType()})");
+				}
+#endif
+			}
+#endif
+		}
+
+		private static void OnSafeAreaOverrideChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+		{
+#if WINUI
+			// VisibleBoundsPadding is disabled for WinUI 3 and up as there's available API for bounds.
+#else
+			if (dependencyObject is FrameworkElement fe)
+			{
+				SafeAreaDetails.GetInstance(fe).OnSafeAreaOverrideChanged((Thickness?)args.OldValue, (Thickness?)args.NewValue);
+			}
+			else
+			{
+#if HAS_UNO // Is building using Uno.UI
+				if (dependencyObject.Log().IsEnabled(LogLevel.Debug))
+				{
+					dependencyObject.Log().LogDebug($"SafeAreaOverride is only supported on FrameworkElement (Found {dependencyObject?.GetType()})");
+				}
+#endif
+			}
+#endif
+		}
+
+		private static void OnInsetModeChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+		{
+#if WINUI
+			// VisibleBoundsPadding is disabled for WinUI 3 and up as there's available API for bounds.
+#else
+			if (dependencyObject is FrameworkElement fe)
+			{
+				SafeAreaDetails.GetInstance(fe).OnInsetModeChanged((InsetMode)args.OldValue, (InsetMode)args.NewValue);
+			}
+			else
+			{
+#if HAS_UNO // Is building using Uno.UI
+				if (dependencyObject.Log().IsEnabled(LogLevel.Debug))
+				{
+					dependencyObject.Log().LogDebug($"Mode is only supported on FrameworkElement (Found {dependencyObject?.GetType()})");
+				}
+#endif
+			}
+#endif
+		}
+
+		/// <summary>
+		/// If false, ApplicationView.VisibleBounds and XamlWindow.Current.Bounds have different aspect ratios (eg portrait vs landscape) which 
+		/// might arise transiently when the screen orientation changes.
+		/// </summary>
+		private static bool AreBoundsAspectRatiosConsistent
+		{
+			get
+			{
+#if WINUI
+				return true;
+#else
+#if HAS_UNO
+				return ApplicationView.GetForCurrentView().VisibleBounds.GetOrientation() == XamlWindow.Current?.Bounds.GetOrientation();
+#else
+				return true;
+#endif
+#endif
+			}
+		}
+
+		private static Thickness GetWindowInsets(Rect? safeAreaOverride = null, bool withSoftInput = false)
+		{
+#if WINUI
+			return new();
```

Comment:
why are we returning `new()`? if the method is not intended to be called in `WINUI` just throw new platformnotsupportedexception

Reply:
This was the behavior within Uno's VisibleBoundsPadding so I just brought it over. I figure we don't want to crash if they run the app for the WinAppSdk head